glean-core/src/upload/mod.rs

```diff
@@ -139,13 +211,29 @@ impl PingUploadManager {
             queue,
             processed_pending_pings,
             directory_manager,
+            rate_limiter: None,
         }
     }
 
     fn has_processed_pings_dir(&self) -> bool {
         self.processed_pending_pings.load(Ordering::SeqCst)
     }
 
+    /// Adds rate limiting capability to this upload manager.
+    ///
+    /// The rate limiter will limit the amount of calls to `get_upload_task` per interval.
```

Comment:
Let's mention explicitly that this resets the state of the rate limiter: e.g. if you sent the maximum amount of pings in the timeframe, then you reset, we start counting again.